### PR TITLE
Charts sso rundeck

### PR DESCRIPTION
**Pull request by Deepak-Routray**
Followup to earlier cisco-sso PR #15 
upstream PR is https://github.com/kubernetes/charts/pull/6641

@josdotso Please review this PR rather than PR #15 
I discussed with @dcwangmit01 on the same. Original PR source repo is deleted so no new push is possible there.
Thanks for your time.

### CODE REVIEW

**Comment by dcwangmit01** [comment]
@Deepak-Routray Closing the old PR and creating this new one makes it nearly impossible to track the changes requested.  It's quadrupling the effort it takes to review this.  I'm certain you could have recovered the old code commits by inspecting the git reflog regardless of branch deletion.  You should still attempt to do so.

Regardless, reviewing now.

File: src/rundeck/templates/configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    framework.ssh.user = {{ .Values.env.RD_USER }}
+    
+    # ssh connection timeout after a specified number of milliseconds.
+    # "0" value means wait forever.
+    framework.ssh.timeout = 0
+    
+    # ----------------------------------------------------------------
+    # Auto generated server UUID
+    # ----------------------------------------------------------------
+    rundeck.server.uuid =
```
Comment (dcwangmit01) [inline src/rundeck/templates/configmap.yaml:53]: In the prior PR you said: "uuid generation is at dockerfile and shell script level. so not present in the chart"

I don't see the Dockerfile anywhere.

**Review by dcwangmit01 — COMMENTED** [review]
At the very least, you should check out the old commits, add the changes on top, and then submit a new PR: https://help.github.com/articles/checking-out-pull-requests-locally/

File: src/rundeck/values.yaml [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+resources:
+  # We usually recommend not to specify default resources and to leave this as a conscious
+  # choice for the user. This also increases chances charts run on environments with little
+  # resources, such as Minikube. If you do want to specify resources, uncomment the following
+  # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
+  #limits:
+  #  cpu: 1
+  #  memory: 1024Mi
+  #requests:
+  #  cpu: 100
```
Comment (dcwangmit01) [inline src/rundeck/values.yaml]: MUST: This is incorrect.  Should be 100m with the "m".

File: src/rundeck/values.yaml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  DB_SCHEMA: rundeck
+  DB_USER: rundeck
+  DB_PASS: rundeck
+  RD_URL: "http://localhost:4440"
+  RD_USER: rundeck
+  RD_PASSWORD: rundeck
+  RD_PROJECT: DEFAULT
+  RUNDECK_INSTALL_DIR: /opt/rundeck
+  RSA_KEY_DIR: /var/lib/rundeck/.ssh
+  JAVA_OPTIONS: -XX:MaxPermSize=256m -Xmx512m -Xms256m -XX:MaxMetaspaceSize=256m -XX:+PrintFlagsFinal
```
Comment (dcwangmit01) [inline src/rundeck/values.yaml]: MUST: Xmx and Xms must be equal

File: src/rundeck/values.yaml [hunk 4]
```diff
@@ -0,0 +1,115 @@
+# Default values for rundeck.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount: 1
+
+image:
+  repository: deepakr11/rundeck
```
Comment (dcwangmit01) [inline src/rundeck/values.yaml:8]: I don't believe you can commit to updating your own docker image in lock step with the community. It'd be better to override the pod spec command line to do things before calling the original entrypoint.

For the values.yaml, use the community image as default. For your own usage, use the helm --set command for your own customizations.

Mount a custom setup script via a configmap. Then override the original entrypoint.  Make it all configurable via values.yaml.

File: src/rundeck/values.yaml [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+#####################################################################
+# Application Configurations
+
+#######################################
+# Pod Spec Overrides
+
+#appname: rundeck
+
+env:
+  SHARED_STORAGE: /rundeck
```
Comment (dcwangmit01) [inline src/rundeck/values.yaml:59]: I don't see sstarcher's comments taken care of
image

**Review by dcwangmit01 — CHANGES_REQUESTED** [review]

File: src/rundeck/templates/configmap.yaml (hunk 1 above)
Comment (Deepak-Routray) [reply src/rundeck/templates/configmap.yaml:53]: Reference of same is there in Chart.yaml
source:
  - https://github.com/Deepak-Routray/rundeck-docker

Relevant Changes here:
https://github.com/Deepak-Routray/rundeck-docker/blob/master/data/bootstrap.sh#L63

File: src/rundeck/values.yaml (hunk 5 above)
Comment (Deepak-Routray) [reply src/rundeck/values.yaml:59]: This particular scenario, we cannot do as we are defining env variables and that should be defined as flat structure. Other places are covered.

**Comment by dcwangmit01** [comment]
Fix the 100m and the Xms/Xmx and I'll approve

File: src/rundeck/values.yaml (hunk 2 above)
Comment (Deepak-Routray) [reply src/rundeck/values.yaml]: Fixed the same.

File: src/rundeck/values.yaml (hunk 3 above)
Comment (Deepak-Routray) [reply src/rundeck/values.yaml]: Fixed the same.

**Review by dcwangmit01 — APPROVED** [review]

**Comment by dcwangmit01** [comment]
@Deepak-Routray Once you resolve these conflicts, please go ahead and merge yourself.

**Comment by rustycl0ck** [comment]
@Deepak-Routray Please fix the README for your OpenAM chart too

**Pull request merged**